frame/support/src/traits/tokens/fungible.rs

```diff
@@ -84,28 +108,74 @@ pub trait Transfer<AccountId>: Inspect<AccountId> {
 	) -> Result<Self::Balance, DispatchError>;
 }
 
-/// Trait for providing a fungible asset which can be reserved.
-pub trait Reserve<AccountId>: Inspect<AccountId> {
+/// Trait for inspecting a fungible asset which can be reserved.
+pub trait InspectReserve<AccountId>: Inspect<AccountId> {
 	/// Amount of funds held in reserve by `who`.
 	fn reserved_balance(who: &AccountId) -> Self::Balance;
+
 	/// Amount of funds held in total by `who`.
 	fn total_balance(who: &AccountId) -> Self::Balance {
 		Self::reserved_balance(who).saturating_add(Self::balance(who))
 	}
+
 	/// Check to see if some `amount` of funds may be reserved on the account of `who`.
 	fn can_reserve(who: &AccountId, amount: Self::Balance) -> bool;
+}
+
+/// Trait for mutating a fungible asset which can be reserved.
+pub trait MutateReserve<AccountId>: InspectReserve<AccountId> + Transfer<AccountId> {
 	/// Reserve some funds in an account.
 	fn reserve(who: &AccountId, amount: Self::Balance) -> DispatchResult;
-	/// Unreserve some funds in an account.
-	fn unreserve(who: &AccountId, amount: Self::Balance) -> DispatchResult;
-	/// Transfer reserved funds into another account.
+
+	/// Unreserve up to `amount` funds in an account.
+	///
+	/// The actual amount unreserved is returned with `Ok`.
+	fn unreserve(who: &AccountId, amount: Self::Balance) -> Result<Self::Balance, DispatchError>;
+
+	/// Transfer up to `amount` funds into another account on a best-effort basis.
+	///
+	/// The actual amount transferred is returned with `Ok`.
 	fn repatriate_reserved(
-		who: &AccountId,
+		source: &AccountId,
+		dest: &AccountId,
 		amount: Self::Balance,
-		status: BalanceStatus,
```

Comment:
You have removed `BalanceStatus` so I would guess you might need to include a `transfer_reserved` function which will move the reserved balance from one person to another.

This could have a default implementation using `repatriate_reserved` + `reserve`

Reply:
I don't think we're using the reserved -> reserved transfer functionality anywhere, which is what `BalanceStatus` here would allow. If it's needed, then I would refactor the calling code.

Reply:
Added `transfer_reserved` to manage this.